cmd/readiness/main.go

```diff
@@ -138,12 +137,18 @@ func findCurrentStep(processStatuses map[string]health.MmsDirectorStatus) *healt
 	return lastStartedStep
 }
 
-func isDeadlocked(status *health.StepStatus) bool {
+// isWaitStep return true is the Agent is currently waiting for something.
+// The 15 seconds explanation:
+//   - The status file is written every 10s but the Agent processes steps independently of it
+//   - In order to avoid reacting on a newly added wait Step (as they can naturally go away), we're giving the Agent
+//     at least 15 sends to spend on that Step.
+//   - This hopefully prevents the Probe from flipping False to True too quickly.
+func isWaitStep(status *health.StepStatus) bool {
 	// Some logic behind 15 seconds: the health status file is dumped each 10 seconds, so we are sure that if the agent
 	// has been in the step for 10 seconds - this means it is waiting for the other hosts, and they are not available
 	fifteenSecondsAgo := time.Now().Add(time.Duration(-15) * time.Second)
-	if contains.String(riskySteps, status.Step) && status.Completed == nil && status.Started.Before(fifteenSecondsAgo) {
-		logger.Infof("Indicated a possible deadlock, status: %s, started at %s but hasn't finished "+
+	if status.IsWaitStep && status.Completed == nil && status.Started.Before(fifteenSecondsAgo) {
```

Comment:
Thanks a lot - I'll perform all the test in both Community and Enterprise.

> I'm not sure what changes that would be in AC that would cause new plans to be generated

This one is very, very tricky. I only managed to artificially test it by handcrafting the changes (and it worked fine). The thing is that under certain circumstances, the Automation Agent might completely recompute the plan. Then, the new plan is appended to the (therefore, it's the last one). So far I've seen only once such situation - someone seemed to delete the automation agent user when rolling out new deployment. However, I failed to recreate this scenario with more recent versions of the Agent.

Reply:
I performed all the test you mentioned. They all seem to be fine and at the same time, the probe seems to properly guard the cluster from rolling multiple members at once.